lib/Mail/DMARC/Report/Receive.pm

```diff
@@ -27,11 +27,12 @@ sub from_imap {
     my $folder = $self->config->{imap}{folder} or croak "no imap folder conf";
     my $a_done = $self->config->{imap}{a_done};
     my $f_done = $self->config->{imap}{f_done};
-    my $port   = $self->get_imap_port();
+    my $port   = $self->config->{imap}{port} // $self->get_imap_port();
```

Comment:
I'd prefer you move this config check into get_imap_port(), so that a single function encapsulates all the logic for which imap port gets chosen.